pkg/engine/update.go

```diff
@@ -184,16 +186,34 @@ func installPlugins(
 	return allPlugins, defaultProviderVersions, nil
 }
 
-func installAndLoadPolicyPlugins(plugctx *plugin.Context, policies []RequiredPolicy) error {
+func installAndLoadPolicyPlugins(plugctx *plugin.Context, policies []RequiredPolicy, localPolicyPackPaths []string,
+	opts *plugin.PolicyAnalyzerOptions) error {
+
+	// Install and load required policy packs.
 	for _, policy := range policies {
 		policyPath, err := policy.Install(context.Background())
 		if err != nil {
 			return err
 		}
 
-		_, err = plugctx.Host.PolicyAnalyzer(tokens.QName(policy.Name()), policyPath)
+		_, err = plugctx.Host.PolicyAnalyzer(tokens.QName(policy.Name()), policyPath, opts)
+		if err != nil {
+			return err
```

Comment:
Nit: I assume it would be helpful to wrap the error and include the policy pack name? Since (worst case scenario) "unable to load policy pack in /tmp/foo/bar" isn't as useful as "loading policy pack 'aws-guard@v3.0.1': unable to load policy pack in '/tmp/foo/bar'"?

Reply:
Sure. Although, this code here hasn't really changed in this PR, so your feedback is orthogonal to the change in the PR. I've just moved the "load local policy packs" below into this block from [step_generator.go](https://github.com/pulumi/pulumi/pull/3612/files#diff-152cf3031402283e2b7aa998a44043cc) to have one place for constructing the options, and the diff makes it look like this is new code.